### DISCUSSION TITLE
Customize template and counter debounce

### DISCUSSION THREAD

**Discussion by aik** [arendst/Tasmota#18293] [Q&A]
(the subject line needs correction as I figured the USER_TEMPLATE problem)
I am making a custom firmware with hardcoded wifi ap and other things. ESP8266 (NodeMCU? a dev board), DHT+VEML607+Counter (rain gauge flipping bucket). Almost there but I am stuck at ~~defining sensors and~~ counter debounce timeout. 

tasmota/user_config_override.h:
```
#define MODULE USER_MODULE
#define USER_TEMPLATE "{\"NAME\":\"WeatherStation\",\"GPIO\":[1,1,384,1,640,608,0,0,1,1216,1,1,1,4704],\"FLAG\":0,\"BASE\":18}"
#define PULSE_COUNTER_DEBOUNCE 1000
```
Of course the PULSE_COUNTER_DEBOUNCE is not there so I added:
```
index f53abb684160..9cf1f760b96f 100644                                  
--- a/tasmota/tasmota_support/settings.ino                               
+++ b/tasmota/tasmota_support/settings.ino                               
@@ -1562,6 +1562,9 @@ void SettingsDelta(void) {                         
     if (Settings->version < 0x0B000003) {  // 11.0.0.3                  
        memcpy(Settings->pulse_timer, Settings->ex_pulse_timer, 16);     
     }                                                                   
+#ifdef PULSE_COUNTER_DEBOUNCE                                           
+    Settings->pulse_counter_debounce = PULSE_COUNTER_DEBOUNCE;    // 5D2
+#endif                                                                  
```

However after resetting the config and flashing my new tasmota.tar.gz, the debounce timeout is not set (the console says it is "0") ~~and SONOFF_BASIC template is selected~~ so I have to change that after flashing the firmware, is there a way to avoid running "CounterDebounce 1000"?

**Comment by aik** [comment]
Should have set pulse_counter_debounce in SettingsDefaultSet2(), not in SettingsDelta(), stupid me. Sorry for the noise, remove the question please.

**Reply by barbudor** [reply]
Always good to keeps such questions for others to come

Another solution which avoid modifying the core source code (which you will have to redo everytime you upgrade the base code unless you propose a definitive change) is to include the "CMND" key in your template such as:
```
..., \"CMND\":\"CounterDebounce 1000\"}"
```

The CMND key acts like a "Backlog" of commands that are executed the 1st time a firwmare is ran on a clean flash

**Reply by aik** [reply]
oh thanks, this CMND is even better!
